django/utils/codegen/__init__.py

```diff
@@ -0,0 +1,23 @@
+def _identity(f):
+    return f
+
+
+def from_codegen(f):
+    """
+    This indicates that the function was gotten from codegen, and
+    should not be directly modified
+    """
+    return f
+
+
+def generate_unasynced_codegen(async_unsafe=False):
```

Comment:
to keep the codegen code simple, we can't do `@generate_unasynced_codegen`, but must do `@generate_unasynced_codegen()`